File: app/src/main/java/ru/nsu/fit/modao/AuthorizationFragment.java
package ru.nsu.fit.modao;

import android.content.IntentFilter;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.NavController;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.TextView;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import ru.nsu.fit.modao.databinding.FragmentAuthorizationBinding;
import ru.nsu.fit.modao.model.UserIn;
import ru.nsu.fit.modao.repository.LoginViewModel;
import ru.nsu.fit.modao.repository.MyApplication;


public class AuthorizationFragment extends Fragment {
    LoginViewModel loginViewModel;
    MyApplication app;
    BottomNavigationView bottomMenu;
    NavController navController;
    FragmentAuthorizationBinding binding;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = FragmentAuthorizationBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }
    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        bottomMenu = getActivity().findViewById(R.id.bottomMenu);
        navController = Navigation.findNavController(view);
        app = (MyApplication) getActivity().getApplication();
        loginViewModel = new ViewModelProvider(this).get(LoginViewModel.class);
        loginViewModel.getUserIn().observe(getViewLifecycleOwner(), new Observer<Long>() {
            @Override
            public void onChanged(Long id) {
                if (id == -1L){
                    binding.tipMessage.setText(R.string.incorrectLoginOrPassword);
                    return;
                }
                if (id == -2L){
                    binding.tipMessage.setText(R.string.serverProblems);
                    return;
                }
                app.setUserID(id);
                bottomMenu.setVisibility(View.VISIBLE);
                NavDirections action = AuthorizationFragmentDirections.actionAuthorizationFragmentToProfileFragment();
                navController.navigate(action);
            }
        });
        binding.buttonLogIn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String log = binding.personLogin.getText().toString();
                String pass = binding.personPassword.getText().toString();
                if (log.equals("") || pass.equals("")){
                    binding.tipMessage.setText(R.string.enterData);
                    return;
                }
                binding.tipMessage.setText(R.string.wait);
                loginViewModel.requestLogin(app.getQuestApi(),
                        new UserIn(log, pass));
            }
        });

        binding.buttonSignUp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                NavDirections action = AuthorizationFragmentDirections.actionAuthorizationFragmentToRegistrationFragment();
                navController.navigate(action);
            }
        });
    }
}

File: app/src/main/java/ru/nsu/fit/modao/fragments/AuthorizationFragment.kt
package ru.nsu.fit.modao.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.findNavController
import com.google.android.material.bottomnavigation.BottomNavigationView
import dagger.hilt.android.AndroidEntryPoint
import ru.nsu.fit.modao.R
import ru.nsu.fit.modao.databinding.FragmentAuthorizationBinding
import ru.nsu.fit.modao.models.Authorization
import ru.nsu.fit.modao.utils.App
import ru.nsu.fit.modao.utils.Constants.Companion.ACCESS_TOKEN
import ru.nsu.fit.modao.utils.Constants.Companion.ID_USER
import ru.nsu.fit.modao.viewmodels.LoginViewModel
import javax.inject.Inject


@AndroidEntryPoint
class AuthorizationFragment : Fragment() {
    private var _binding: FragmentAuthorizationBinding? = null
    private val binding get() = _binding!!

    @Inject
    lateinit var app: App
    private val loginViewModel: LoginViewModel by viewModels()
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentAuthorizationBinding.inflate(inflater, container, false)
        return binding.root

    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        loginViewModel.message.observe(viewLifecycleOwner) {
            binding.tipMessage.text = it
        }
        loginViewModel.token.observe(viewLifecycleOwner) {
            login(it)
        }
        binding.buttonLogIn.setOnClickListener {
            binding.tipMessage.setText(R.string.wait)
            loginViewModel.login(
                login = binding.personLogin.text.toString(),
                password = binding.personPassword.text.toString()
            )
        }
        binding.buttonSignUp.setOnClickListener {
            findNavController().navigate(AuthorizationFragmentDirections
                .actionAuthorizationFragmentToRegistrationFragment())
        }
    }
    private fun login(it: Authorization){
        app.userId = it.id
        app.accessToken = it.accessToken
        app.refreshToken = it.refreshToken
        val edit = app.encryptedSharedPreferences.edit()
        edit.putString(ACCESS_TOKEN, it.accessToken).putLong(ID_USER, it.id).apply()
        activity?.findViewById<BottomNavigationView>(R.id.bottomMenu)?.visibility = View.VISIBLE
        findNavController().navigate(AuthorizationFragmentDirections.actionAuthorizationFragmentToProfileFragment())
    }
}